src/os/shared/src/osapi-binsem.c

```diff
@@ -157,14 +157,8 @@ int32 OS_BinSemDelete (uint32 sem_id)
    {
       return_code = OS_BinSemDelete_Impl(local_id);
 
-      /* Free the entry in the master table now while still locked */
-      if (return_code == OS_SUCCESS)
```

Comment:
return_code is set on line 158 and not checked with this deletion. It's set again on 161.

Reply:
The return code is checked inside the `OS_ObjectIdFinalizeDelete` function.  There is cleanup to do for both success and failure cases.  (This is a mirror image of the way the allocation/creation works too).